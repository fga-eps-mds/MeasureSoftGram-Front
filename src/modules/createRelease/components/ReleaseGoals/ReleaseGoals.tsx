import React from 'react';

import Equalizer from '@modules/createRelease/components/Equalizer';

import { useCreateReleaseContext } from '@modules/createRelease/context/useCreateRelease';
import {
  Box,
  Button,
  Checkbox,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Link,
  Typography
} from '@mui/material';
import * as Styles from './styles';
import useDynamicBalance from './hook/useDynamicBalance';

export default function ReleaseGoals() {
  const { releaseInfoForm } = useCreateReleaseContext();
  const { characteristics, endDate, name, startDate } = releaseInfoForm;
  const { open, allowDynamicBalance, handleChange, handleClose, handleConfirm } = useDynamicBalance();

  return (
    <>
      <Styles.Header>
        <h1>Balancear a Meta de Qualidade</h1>
        <p>
          Nesta etapa, é possível estabelecer metas para as características que serão observadas na release. É
          importante ressaltar que essas características de qualidade estão inter-relacionadas, ou seja, ao modificar a
          meta de uma característica, as demais também serão afetadas.
        </p>
        <p>
          Ao clicar em 'Permitir o balanceamento dinâmico', as alterações feitas em uma característica não afetarão
          automaticamente as demais, permitindo uma flexibilidade maior no planejamento da release.
        </p>
        <Box>
          <Typography color="#FF4646">
            <Checkbox checked={allowDynamicBalance} onChange={handleChange} name="allowDynamicBalanceCheckbox" />
            Permitir o balanceamento dinâmico
          </Typography>
        </Box>
        <Dialog open={open} onClose={handleClose}>
          <DialogTitle>Permitir o balanceamento dinâmico</DialogTitle>
          <DialogContent>
            <Typography color="#FF4646">
              O balanceamento das metas funciona com base em uma matriz de relacionamento entre as características de
              qualidade{' '}
              <Link
                href="https://www.sce.carleton.ca/faculty/wainer/papers/art%253A10.1007%252Fs13198-016-0546-8.pdf"
                underline="hover"
                target="_blank"
              >
                (link)
              </Link>
              . Ao permitir o balanceamento dinâmico, o sistema faz com que essas relações sejam ignoradas, dessa forma,
              alguns objetivos definidos podem ser inalcançáveis. Deseja continuar mesmo assim?
            </Typography>
          </DialogContent>
          <DialogActions>
            <Button onClick={handleClose}>Cancelar</Button>
            <Button onClick={handleConfirm} color="primary" name="confirmButton">
              Confirmar
            </Button>
          </DialogActions>
        </Dialog>
      </Styles.Header>

      <Styles.Body>
        <div>
          <strong>{name}</strong>
          <p>
            {startDate} - {endDate}
          </p>
        </div>

        <Equalizer selectedCharacteristics={characteristics} allowDynamicBalance={allowDynamicBalance} />
      </Styles.Body>
    </>
  );
}
