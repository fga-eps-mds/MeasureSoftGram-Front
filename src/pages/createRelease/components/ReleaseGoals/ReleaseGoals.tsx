import React from 'react';

import Equalizer from '@pages/createRelease/components/Equalizer';

import { useCreateReleaseContext } from '@pages/createRelease/context/useCreateRelease';
import * as Styles from './styles';

export default function ReleaseGoals() {
  const { releaseInfoForm } = useCreateReleaseContext();
  const { characteristics, endDate, name, startDate } = releaseInfoForm

  return (
    <>
      <Styles.Header>
        <h1>Balancear a Meta de Qualidade</h1>
        <p>
          Mini explicação como funciona as metas da release
          (Configuração das caracteristicas,
          valor de uma caracteristica pode influenciar em outra, etc)
        </p>
      </Styles.Header>

      <Styles.Body>
        <div>
          <strong>{name}</strong>
          <p>{startDate} - {endDate}</p>
        </div>

        <Equalizer selectedCharacteristics={characteristics}/>
      </Styles.Body>
    </>
  )
}
