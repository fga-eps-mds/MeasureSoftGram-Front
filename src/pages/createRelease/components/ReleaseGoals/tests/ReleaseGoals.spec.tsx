import '@testing-library/jest-dom';

import React from 'react';
import { render } from '@testing-library/react';

import ReleaseGoals from '../ReleaseGoals';

jest.mock('@pages/createRelease/context/useCreateRelease', () => ({
  useCreateReleaseContext: () => ({
    releaseInfoForm: {
      characteristics: [ "reliability", "maintainability" ],
      endDate: "2022-09-25",
      name: "asdasd",
      startDate: "2022-09-18",
    }
  })
}));

describe('<ReleaseGoals />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Snapshot', () => {
    it('Deve corresponder ao Snapshot', () => {
      const tree = render(<ReleaseGoals/>)
      expect(tree).toMatchSnapshot();
    });
  });
});
