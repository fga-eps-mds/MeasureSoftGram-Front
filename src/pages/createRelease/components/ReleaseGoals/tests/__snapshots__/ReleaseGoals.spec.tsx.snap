// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReleaseGoals /> Snapshot Deve corresponder ao Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__Header-sc-15quhu7-0 dsjdCx"
      >
        <h1>
          Balancear a Meta de Qualidade
        </h1>
        <p>
          Mini explicação como funciona as metas da release (Configuração das caracteristicas, valor de uma caracteristica pode influenciar em outra, etc)
        </p>
      </div>
      <div
        class="styles__Body-sc-15quhu7-1 fsQjbk"
      >
        <div>
          <strong>
            asdasd
          </strong>
          <p>
            2022-09-18
             - 
            2022-09-25
          </p>
        </div>
        <div
          class="MuiBox-root css-ckc6o3"
        >
          <div
            class="MuiBox-root css-t0lfuy"
          />
          <div
            class="styles__Labels-sc-1fexhzx-0 kyDTDD"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__Header-sc-15quhu7-0 dsjdCx"
    >
      <h1>
        Balancear a Meta de Qualidade
      </h1>
      <p>
        Mini explicação como funciona as metas da release (Configuração das caracteristicas, valor de uma caracteristica pode influenciar em outra, etc)
      </p>
    </div>
    <div
      class="styles__Body-sc-15quhu7-1 fsQjbk"
    >
      <div>
        <strong>
          asdasd
        </strong>
        <p>
          2022-09-18
           - 
          2022-09-25
        </p>
      </div>
      <div
        class="MuiBox-root css-ckc6o3"
      >
        <div
          class="MuiBox-root css-t0lfuy"
        />
        <div
          class="styles__Labels-sc-1fexhzx-0 kyDTDD"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
